maps/_map.dm

```diff
@@ -176,6 +180,28 @@ var/global/list/accessable_z_levels = list()
 	movementJammed = 1
 	base_turf = /turf/unsimulated/beach/sand
 
+/datum/zLevel/snow
+	name = "snow"
+	base_turf = /turf/unsimulated/floor/snow
+	movementChance = ZLEVEL_BASE_CHANCE * ZLEVEL_SPACE_MODIFIER
+
+/datum/zLevel/snow/post_mapload()
+	var/lake_density = rand(2,8)
+	for(var/i = 0 to lake_density)
+		var/turf/T = locate(rand(0, world.maxx),rand(0, world.maxy), z)
+		if(!istype(T, base_turf))
+			continue
+		var/generator = pick(typesof(/obj/structure/radial_gen/cellular_automata/ice))
+		new generator(T)
+
+	var/tree_density = rand(25,45)
+	for(var/i = 0 to tree_density)
+		var/turf/T = locate(rand(0,world.maxx),rand(0, world.maxy), z)
```

Comment:
```locate(0, y, z)``` will just spawn your thing in nullspace, so `rand(1, world.maxx)` is probably better